Firestore/Swift/Tests/Integration/QueryIntegrationTests.swift

```diff
@@ -0,0 +1,208 @@
+      [Filter.whereField("a", isEqualTo: 2),
+       Filter.whereField("b", isEqualTo: 1)]
+    )
+    checkOnlineAndOfflineQuery(collRef.whereFilter(filter8).limit(to: 1)
+      .order(by: "a"),
+      matchesResult: ["doc5"])
+
+    // Test with limits (explicit order by DESC): (a==2) || (b == 1) ORDER BY a LIMIT_TO_LAST 1
+    let filter9 = Filter.orFilter(
+      [Filter.whereField("a", isEqualTo: 2),
+       Filter.whereField("b", isEqualTo: 1)]
+    )
+    checkOnlineAndOfflineQuery(collRef.whereFilter(filter9).limit(toLast: 1)
+      .order(by: "a"),
+      matchesResult: ["doc2"])
+
+    // Test with limits without orderBy (the __name__ ordering is the tie breaker).
+    let filter10 = Filter.orFilter(
+      [Filter.whereField("a", isEqualTo: 2),
+       Filter.whereField("b", isEqualTo: 1)]
+    )
+    checkOnlineAndOfflineQuery(collRef.whereFilter(filter10).limit(to: 1),
+                               matchesResult: ["doc2"])
+  }
+
+  func testOrQueriesWithInAndNotIn() throws {
+    let collRef = collectionRef(
+      withDocuments: ["doc1": ["a": 1, "b": 0],
+                      "doc2": ["b": 1],
+                      "doc3": ["a": 3, "b": 2],
+                      "doc4": ["a": 1, "b": 3],
+                      "doc5": ["a": 1],
+                      "doc6": ["a": 2]]
+    )
+
+    // a==2 || b in [2,3]
+    let filter1 = Filter.orFilter(
+      [Filter.whereField("a", isEqualTo: 2),
+       Filter.whereField("b", in: [2, 3])]
+    )
+    checkOnlineAndOfflineQuery(collRef.whereFilter(filter1),
+                               matchesResult: ["doc3", "doc4", "doc6"])
+
+    // a==2 || b not-in [2,3]
+    // Has implicit orderBy b.
+    let filter2 = Filter.orFilter(
+      [Filter.whereField("a", isEqualTo: 2),
+       Filter.whereField("b", notIn: [2, 3])]
+    )
+    checkOnlineAndOfflineQuery(collRef.whereFilter(filter2),
+                               matchesResult: ["doc1", "doc2"])
+  }
+
+  func testOrQueriesWithArrayMembership() throws {
+    let collRef = collectionRef(
+      withDocuments: ["doc1": ["a": 1, "b": [0]],
+                      "doc2": ["b": 1],
+                      "doc3": ["a": 3, "b": [2, 7]],
+                      "doc4": ["a": 1, "b": [3, 7]],
+                      "doc5": ["a": 1],
+                      "doc6": ["a": 2]]
+    )
+
+    // a==2 || b array-contains 7
+    let filter1 = Filter.orFilter(
+      [Filter.whereField("a", isEqualTo: 2),
+       Filter.whereField("b", arrayContains: 7)]
+    )
+    checkOnlineAndOfflineQuery(collRef.whereFilter(filter1),
+                               matchesResult: ["doc3", "doc4", "doc6"])
+
+    // a==2 || b array-contains-any [0, 3]
+    let filter2 = Filter.orFilter(
+      [Filter.whereField("a", isEqualTo: 2),
+       Filter.whereField("b", arrayContainsAny: [0, 3])]
+    )
+    checkOnlineAndOfflineQuery(collRef.whereFilter(filter2),
+                               matchesResult: ["doc1", "doc4", "doc6"])
+  }
+}
```

Comment:
TODO: Add Swift test for in Filter

Reply:
I _think_ you've already added all the tests. Isn't that right?

Reply:
Yes, forgot to resolve this